src/Admin/AdminUserRepository.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EtoA\Admin;
+
+use EtoA\Core\AbstractRepository;
+
+class AdminUserRepository extends AbstractRepository
+{
+    function count(): int
+    {
+        return (int) $this->createQueryBuilder()
+            ->select("COUNT(*)")
+            ->from('admin_users')
+            ->execute()
+            ->fetchOne();
+    }
+
+    public function find(int $id): ?AdminUser
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('admin_users')
+            ->where('user_id = ?')
+            ->setParameter(0, $id)
+            ->execute()
+            ->fetchAssociative();
+        return !is_bool($data) ? $this->createObject($data) : null;
+    }
+
+    public function findOneByNick(string $nick)
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('admin_users')
+            ->where('LCASE(user_nick) = LCASE(?)')
+            ->setParameter(0, $nick)
+            ->execute()
+            ->fetchAssociative();
+        return !is_bool($data) ? $this->createObject($data) : null;
+    }
+
+    public function findAll(): array
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('admin_users')
+            ->orderBy('user_nick')
+            ->execute()
+            ->fetchAllAssociative();
+        return collect($data)
+            ->map(fn ($arr) => $this->createObject($arr))
+            ->toArray();
+    }
+
+    public function findAllAsList(): array
+    {
+        return $this->createQueryBuilder()
+            ->select("user_id", 'user_nick')
+            ->from('admin_users')
+            ->orderBy('user_nick')
+            ->execute()
+            ->fetchAllKeyValue();
+    }
+
+    private function createObject(array $data): AdminUser
+    {
+        $adminUser = new AdminUser();
+        $adminUser->id = (int) $data['user_id'];
```

Comment:
This is something I would move to a constructor or to a named constructor to make sure that the object is fully initialized after the constructor call. Also might remove the need for default values in the class

Reply:
So basically moving the mapping from the DB array into the item class?

Something like:

```
class AdminUser
{
...
    public static function createFromArray(array $data): AdminUser
    {
        $adminUser = new AdminUser();
        $adminUser->id = (int) $data['user_id'];
        $adminUser->name = $data['user_name'];
...
        return $adminUser;
    }
...
}
```

Reply:
yes

Reply:
Done in b0d25fe